sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/s3/S3FileSystem.java

```diff
@@ -515,15 +516,17 @@ CompleteMultipartUploadResponse multipartCopy(
     if (objectSize == 0) {
       final UploadPartCopyRequest uploadPartCopyRequest =
           UploadPartCopyRequest.builder()
-              .bucket(sourcePath.getBucket())
-              .key(sourcePath.getKey())
+              .bucket(destinationPath.getBucket())
+              .key(destinationPath.getKey())
```

Comment:
It was suggested on the jira to use the non deprecated api, so I'll make an update shortly

Reply:
this has been done.